scarb/src/ui/mod.rs

```diff
@@ -37,22 +37,37 @@ impl Default for Verbosity {
     }
 }
 
-impl Verbosity {
-    fn as_level_filter(&self) -> LevelFilter {
+impl Display for Verbosity {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
-            Self::Quiet => LevelFilter::OFF,
-            Self::Normal => LevelFilter::ERROR,
-            Self::Verbose => LevelFilter::TRACE,
+            Self::Quiet => write!(f, "quiet"),
+            Self::Normal => write!(f, "normal"),
+            Self::Verbose => write!(f, "verbose"),
         }
     }
+}
+
+impl FromStr for Verbosity {
+    type Err = anyhow::Error;
+
+    fn from_str(s: &str) -> Result<Self> {
+        if s.is_empty() {
+            bail!("empty string cannot be used as verbosity level");
+        }
+
+        match s {
+            "quiet" => Ok(Verbosity::Quiet),
+            "normal" => Ok(Verbosity::Normal),
+            "verbose" => Ok(Verbosity::Verbose),
+            _ => bail!("invalid verbosity level: {}", s),
```

Comment:
clippy should yell with this:

```suggestion
            _ => bail!("invalid verbosity level: {s}"),
```